src/mimeparser.rs

```diff
@@ -909,14 +911,20 @@ pub(crate) struct Report {
     additional_message_ids: Vec<String>,
 }
 
-fn parse_message_id(field: &str) -> Option<String> {
-    if let Ok(addrs) = mailparse::addrparse(field) {
-        // Assume the message id is a single id in the form of <id>
-        if let mailparse::MailAddr::Single(mailparse::SingleInfo { ref addr, .. }) = addrs[0] {
-            return Some(addr.clone());
+pub(crate) fn parse_message_id(value: &str) -> crate::error::Result<String> {
+    let ids = mailparse::msgidparse(value)
+        .map_err(|err| format_err!("failed to parse message id {:?}", err))?;
+
+    if ids.len() == 1 {
+        let id = &ids[0];
+        if id.starts_with('<') && id.ends_with('>') {
+            Ok(id.chars().skip(1).take(id.len() - 2).collect())
```

Comment:
looks a bit complicated -- i take it that something like `id[1..id.len()-1].clone()` or so does not work?

Reply:
Indexing operates on byte level, it is unsafe.